### ISSUE TITLE
RF05: Incorrectly detects table as identifier in BigQuery `DELETE FROM` statement 

### ISSUE THREAD

**Issue by dmohns** [sqlfluff/sqlfluff#4646]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Consider the following SQL

```sql
DELETE FROM `project_id.dataset_id.target_name`
WHERE TRUE
;
```

yields:

```
L:   1 | P:  13 | RF05 | Do not use special characters in identifiers.
                       | [references.special_chars]
```

However, `project_id.dataset_id.target_name` is not an identifier but a table reference. 

### Expected Behaviour

SQLFluff should pass.

### Observed Behaviour

see above

### How to reproduce

see above

### Dialect

BigQuery

### Version

2.0.2

### Configuration

default

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
